check_ldap_monitor_389ds.pl

```diff
@@ -0,0 +1,607 @@
+    $ldap_scope  ||= "base";
+    $ldap_base   ||= "cn=Monitor";
+    $ldap_attribute = "totalconnections";
+    $type_defined   = 1;
+}
+if ( $type =~ /dtablesize/i ) {
+    $type_string = "dtablesize";
+    $ldap_filter ||= "(objectClass=*)";
+    $ldap_scope  ||= "base";
+    $ldap_base   ||= "cn=Monitor";
+    $ldap_attribute = "dtablesize";
+    $type_defined   = 1;
+}
+if ( $type =~ /readwaiters/i ) {
+    $type_string = "readwaiters";
+    $ldap_filter ||= "(objectClass=*)";
+    $ldap_scope  ||= "base";
+    $ldap_base   ||= "cn=Monitor";
+    $ldap_attribute = "readwaiters";
+    $type_defined   = 1;
+}
+if ( $type =~ /opsinitiated/i ) {
+    $type_string = "opsinitiated";
+    $ldap_filter ||= "(objectClass=*)";
+    $ldap_scope  ||= "base";
+    $ldap_base   ||= "cn=Monitor";
+    $ldap_attribute = "opsinitiated";
+    $type_defined   = 1;
+}
+if ( $type =~ /opscompleted/i ) {
+    $type_string = "opscompleted";
+    $ldap_filter ||= "(objectClass=*)";
+    $ldap_scope  ||= "base";
+    $ldap_base   ||= "cn=Monitor";
+    $ldap_attribute = "opscompleted";
+    $type_defined   = 1;
+}
+if ( $type =~ /entriessent/i ) {
+    $type_string = "entriessent";
+    $ldap_filter ||= "(objectClass=*)";
+    $ldap_scope  ||= "base";
+    $ldap_base   ||= "cn=Monitor";
+    $ldap_attribute = "entriessent";
+    $type_defined   = 1;
+}
+if ( $type =~ /bytessent/i ) {
+    $type_string = "bytessent";
+    $ldap_filter ||= "(objectClass=*)";
+    $ldap_scope  ||= "base";
+    $ldap_base   ||= "cn=Monitor";
+    $ldap_attribute = "bytessent";
+    $type_defined   = 1;
+}
+if ( $type =~ /currenttime/i ) {
+    $type_string = "currenttime";
+    $ldap_filter ||= "(objectClass=*)";
+    $ldap_scope  ||= "base";
+    $ldap_base   ||= "cn=Monitor";
+    $ldap_attribute = "currenttime";
+    $type_defined   = 1;
+}
+if ( $type =~ /starttime/i ) {
+    $type_string = "starttime";
+    $ldap_filter ||= "(objectClass=*)";
+    $ldap_scope  ||= "base";
+    $ldap_base   ||= "cn=Monitor";
+    $ldap_attribute = "starttime";
+    $type_defined   = 1;
+}
+if ( $type =~ /nbackends/i ) {
+    $type_string = "nbackends";
+    $ldap_filter ||= "(objectClass=*)";
+    $ldap_scope  ||= "base";
+    $ldap_base   ||= "cn=Monitor";
+    $ldap_attribute = "nbackends";
+    $type_defined   = 1;
+}
+unless ($type_defined) {
+    print "Type $type is not known.\n";
+    exit $ERRORS{'UNKNOWN'};
+}
+
+# Request LDAP
+my $value;
+( $errorcode, $value ) =
+  &get_value( $ldap_server, $ldap_base, $ldap_scope, $ldap_filter,
+    $ldap_attribute );
+if ($errorcode) {
+    print "LDAP search failed: $value.\n";
+    exit $ERRORS{'UNKNOWN'};
+}
+
+#==========================================================================
+# Exit with Nagios codes
+#==========================================================================
+
+# Prepare PerfParse data
+my $perfparse = " ";
+if ($perf_data) {
+    $perfparse = "|'$type'=" . $value . ";$warning;$critical";
+}
```

Comment:
If $warning and $critical are not defined, you should set them to 0 to respect perfparse format. Maybe do this after the test to check their value.

Reply:
Does ee4e30945b32ab81d1e890e46b62955c31d1fb5c looks good to you ?

Reply:
Fixed typo in df0293f58f5571e1a15b426ac3ab5e0c07276425